### PR TITLE
Cumulus-687 Retrieve the Cumulus stdout logs for a workflow execution via the API

### PR DESCRIPTION
**Pull request by kkelly51**
As a DAAC operator, I want to retrieve the Cumulus logs for a workflow via the API so I may troubleshoot problems

Addresses [CUMULUS-687](https://bugs.earthdata.nasa.gov/browse/CUMULUS-687)

## Changes

* Updated version of CMA js
* Includes task name and task version in logs
* Added integration tests for the "logs" endpoint

### CODE REVIEW

File: example/spec/testAPI/APISuccessSpec.js [hunk 1]
```diff
@@ -181,5 +181,22 @@ describe('The Cumulus API', () => {
...
+      const logs = await apiTestUtils.getLogs({ prefix: config.stackName });
+      expect(logs).not.toBe(undefined);
+      expect(logs.results.length).toEqual(10);
+    });
+
+    it('returns logs with taskName included', async () => {
+      const logs = await apiTestUtils.getLogs({ prefix: config.stackName });
+      logs.results.forEach((log) => {
+        if ((!log.message.includes('END')) && (!log.message.includes('REPORT')) && (!log.message.includes('START'))) {
+          expect(log.sender).not.toBe(undefined);
```
Comment (jennyhliu) [inline example/spec/testAPI/APISuccessSpec.js]: Can add tests to retrieve the logs based on the following?
1. workflow execution arn/name
2. step name, e. g. CmrStep

File: example/spec/testAPI/APISuccessSpec.js [hunk 2]
```diff
@@ -181,5 +181,22 @@ describe('The Cumulus API', () => {
...
   });
+
+  describe('logs endpoint', () => {
+    it('returns the execution logs', async () => {
+      const logs = await apiTestUtils.getLogs({ prefix: config.stackName });
+      expect(logs).not.toBe(undefined);
+      expect(logs.results.length).toEqual(10);
+    });
+
+    it('returns logs with taskName included', async () => {
```
Comment (jennyhliu) [inline example/spec/testAPI/APISuccessSpec.js]: Can add tests to retrieve the logs based on the following?
1. workflow execution arn/name
2. step name, e. g. CmrStep

File: tasks/discover-granules/package.json [hunk 3]
```diff
@@ -38,7 +38,7 @@
   "license": "Apache-2.0",
   "dependencies": {
     "@cumulus/common": "^1.8.0",
-    "@cumulus/cumulus-message-adapter-js": "^1.0.1",
+    "@cumulus/cumulus-message-adapter-js": "^1.0.2",
```
Comment (mennovandiermen) [inline tasks/discover-granules/package.json]: These versions need to be updated to `^1.0.3`

File: packages/api/endpoints/logs.js [hunk 4]
```diff
@@ -14,12 +19,35 @@ function list(event, cb) {
   });
 }
 
+/**
+ * Query logs from a single function.
```
Comment (mennovandiermen) [inline packages/api/endpoints/logs.js]: 'single workflow execution' would be more accurate.

File: packages/api/es/search.js [hunk 5]
```diff
@@ -173,7 +173,7 @@ class BaseSearch {
           term: {
             _id: id
           }
-        }
+        }
```
Comment (mennovandiermen) [inline packages/api/es/search.js]: trailing space

File: packages/api/config/api_default.yml [hunk 6]
```diff
@@ -309,6 +309,10 @@ ApiLogsDefault:
       method: get
       cors: true
       api: backend
+    - path: logs/{executionName}
+      method: get
+      cors: true
+      api: backend
```
Comment (mennovandiermen) [inline packages/api/config/api_default.yml]: cool, convenient 👍

File: packages/api/es/indexer.js [hunk 7]
```diff
@@ -62,6 +62,8 @@ async function indexLog(esClient, payloads, index = defaultIndexAlias, type = 'l
     catch (e) {
       record = {
         message: p.message.trim(),
+        sender: p.sender,
+        executions: p.executions,
```
Comment (mennovandiermen) [inline packages/api/es/indexer.js]: needs to index `version`

**Pull request merged**